Index: linux-liquorix/include/linux/mm.h
===================================================================
--- linux-liquorix.orig/include/linux/mm.h
+++ linux-liquorix/include/linux/mm.h
@@ -192,7 +192,7 @@ static inline void __mm_zero_struct_page
  * that.
  */
 #define MAPCOUNT_ELF_CORE_MARGIN	(5)
-#define DEFAULT_MAX_MAP_COUNT	(INT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
+#define DEFAULT_MAX_MAP_COUNT	(USHRT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
 
 extern int sysctl_max_map_count;
 
