Index: linux-liquorix/include/linux/mm.h
===================================================================
--- linux-liquorix.orig/include/linux/mm.h
+++ linux-liquorix/include/linux/mm.h
@@ -191,7 +191,7 @@ static inline void __mm_zero_struct_page
  * that.
  */
 #define MAPCOUNT_ELF_CORE_MARGIN	(5)
-#define DEFAULT_MAX_MAP_COUNT	(USHRT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
+#define DEFAULT_MAX_MAP_COUNT	(INT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
 
 extern int sysctl_max_map_count;
 
Index: linux-liquorix/kernel/sysctl.c
===================================================================
--- linux-liquorix.orig/kernel/sysctl.c
+++ linux-liquorix/kernel/sysctl.c
@@ -2147,7 +2147,7 @@ static struct ctl_table vm_table[] = {
 		.procname	= "max_map_count",
 		.data		= &sysctl_max_map_count,
 		.maxlen		= sizeof(sysctl_max_map_count),
-		.mode		= 0644,
+		.mode		= 0444,
 		.proc_handler	= proc_dointvec_minmax,
 		.extra1		= SYSCTL_ZERO,
 	},
